src/protocols/types/SurfaceState.hpp

```diff
@@ -39,7 +60,9 @@ struct SSurfaceState {
     Vector2D offset;
 
     // for xdg_shell resizing
-    Vector2D ackedSize;
+    Vector2D                              ackedSize;
+
+    std::vector<SP<SSurfaceStateFrameCB>> callbacks;
```

Comment:
callbacks belong to core, probably a good idea to just have a `std::vector<SP<CWLCallbackResource>> m_callbacks` in CWLCompositor